canonical-web-and-design/maas-ui
# feat(ui): separate KVM table into LXD and virsh KVMs

## Done

- Renamed `KVMListTable` to `VirshTable` and removed some data/columns that won't be needed anymore
- Copied `VirshTable` over to `LxdTable` with some minor changes to work with grouped LXD KVMs
- Moved the cell components up a level since most of them will be shared between the two tables
- Moved data fetching up into the parent `KVMList` component

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and check that LXD and virsh KVMs each have their own table
- Check that sorting still works in the virsh table (it'll have to be handled differently with the LXD table since it'll be grouped)

## Fixes

Fixes #2221
